src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      result.push({ ...Object.assign(copyState, action.extraData) });
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+
+      result.push({ ...copyState });
+    }
+
+    if (action.type === 'clear') {
+      for (const key in copyState) {
+        delete copyState[key];
+      }
+
+      result.push({ ...copyState });
+    }
+  }
```

Comment:
[DONT REPEAT YOURSELF] - You are pushing to the result array in every case. It is better to do this once after the switch statement to avoid code repetition.